plugins/inputs/docker_log/docker_log.go

```diff
@@ -0,0 +1,430 @@
+  ## Note that an empty array for both will include all labels as tags
+  # docker_label_include = []
+  # docker_label_exclude = []
+
+  ## Optional TLS Config
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = false
+`
+
+const (
+	defaultEndpoint = "unix:///var/run/docker.sock"
+
+	// Maximum bytes of a log line before it will be split, size is mirroring
+	// docker code:
+	// https://github.com/moby/moby/blob/master/daemon/logger/copier.go#L21
+	maxLineBytes = 16 * 1024
+)
+
+var (
+	containerStates = []string{"created", "restarting", "running", "removing", "paused", "exited", "dead"}
+)
+
+type DockerLogs struct {
+	Endpoint              string            `toml:"endpoint"`
+	FromBeginning         bool              `toml:"from_beginning"`
+	Timeout               internal.Duration `toml:"timeout"`
+	LabelInclude          []string          `toml:"docker_label_include"`
+	LabelExclude          []string          `toml:"docker_label_exclude"`
+	ContainerInclude      []string          `toml:"container_name_include"`
+	ContainerExclude      []string          `toml:"container_name_exclude"`
+	ContainerStateInclude []string          `toml:"container_state_include"`
+	ContainerStateExclude []string          `toml:"container_state_exclude"`
+
+	tlsint.ClientConfig
+
+	newEnvClient func() (Client, error)
+	newClient    func(string, *tls.Config) (Client, error)
+
+	client          Client
+	labelFilter     filter.Filter
+	containerFilter filter.Filter
+	stateFilter     filter.Filter
+	opts            types.ContainerListOptions
+	wg              sync.WaitGroup
+	mu              sync.Mutex
+	containerList   map[string]context.CancelFunc
+}
+
+func (d *DockerLogs) Description() string {
+	return "Read logging output from the Docker engine"
+}
+
+func (d *DockerLogs) SampleConfig() string {
+	return sampleConfig
+}
+
+func (d *DockerLogs) Init() error {
+	var err error
+	if d.Endpoint == "ENV" {
+		d.client, err = d.newEnvClient()
+		if err != nil {
+			return err
+		}
+	} else {
+		tlsConfig, err := d.ClientConfig.TLSConfig()
+		if err != nil {
+			return err
+		}
+		d.client, err = d.newClient(d.Endpoint, tlsConfig)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Create filters
+	err = d.createLabelFilters()
+	if err != nil {
+		return err
+	}
+	err = d.createContainerFilters()
+	if err != nil {
+		return err
+	}
+	err = d.createContainerStateFilters()
+	if err != nil {
+		return err
+	}
+
+	filterArgs := filters.NewArgs()
+	for _, state := range containerStates {
+		if d.stateFilter.Match(state) {
+			filterArgs.Add("status", state)
+		}
+	}
+
+	if filterArgs.Len() != 0 {
+		d.opts = types.ContainerListOptions{
+			Filters: filterArgs,
+		}
+	}
+
+	return nil
+}
+
+func (d *DockerLogs) addToContainerList(containerId string, cancel context.CancelFunc) error {
```

Comment:
Perhaps change all `containerId` to `containerID` to match other places in the code and go style?